components/camel-ai/camel-torchserve/src/main/java/org/apache/camel/component/torchserve/TorchServeEndpoint.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+
+@UriEndpoint(firstVersion = "4.9.0", scheme = "torchserve", title = "TorchServe",
+             syntax = "torchserve:api/operation", producerOnly = true,
+             category = { Category.AI }, headersClass = TorchServeConstants.class)
+public class TorchServeEndpoint extends DefaultEndpoint {
+
+    @UriPath(enums = "inference,management,metrics", description = "The TorchServe API")
+    @Metadata(required = true)
+    private final String api;
+
+    @UriPath(enums = "ping,predictions,explanations,register,scale-worker,describe,unregister,list,set-default,metrics",
+             description = "The API operation")
+    @Metadata(required = true)
+    private final String operation;
+
+    @UriParam
+    private TorchServeConfiguration configuration;
+
+    private final TorchServeClient client;
+
+    public TorchServeEndpoint(String uri, TorchServeComponent component, String path,
+                              TorchServeConfiguration configuration) {
+        super(uri, component);
+        String[] parts = extractPath(path);
+        this.api = parts[0];
+        this.operation = parts[1];
+        this.configuration = configuration;
+        this.client = createClient(configuration);
+    }
+
+    private static String[] extractPath(String path) {
+        if (path == null) {
+            throw new IllegalArgumentException("path must not be null");
+        }
+        String[] parts = path.split("/");
+        if (parts.length != 2) {
+            throw new IllegalArgumentException("path must be <api>/<operation>: " + path);
+        }
+        return parts;
+    }
+
+    private static TorchServeClient createClient(TorchServeConfiguration configuration) {
+        TorchServeClient.Builder builder = TorchServeClient.builder();
+        // Inference
+        if (configuration.getInferenceKey() != null) {
+            builder.inferenceKey(configuration.getInferenceKey());
+        }
+        if (configuration.getInferenceAddress() != null) {
+            builder.inferenceAddress(configuration.getInferenceAddress());
+        } else {
+            builder.inferencePort(configuration.getInferencePort());
+        }
+        // Management
+        if (configuration.getManagementKey() != null) {
+            builder.managementKey(configuration.getManagementKey());
+        }
+        if (configuration.getManagementAddress() != null) {
+            builder.managementAddress(configuration.getManagementAddress());
+        } else {
+            builder.managementPort(configuration.getManagementPort());
+        }
+        // Metrics
+        if (configuration.getMetricsAddress() != null) {
+            builder.metricsAddress(configuration.getMetricsAddress());
+        } else {
+            builder.metricsPort(configuration.getMetricsPort());
+        }
+        return builder.build();
```

Comment:
If the build internally setup some thread pool / networking etc then its better to move this into doInit to better support quarkus build time optimization and also tooling such as jbang exports and its also what other components does.